lib/WTSI/NPG/Data/BamDeletion.pm

```diff
@@ -0,0 +1,325 @@
+    close $md5_fh ||
+          $self->logcroak(q[Failed to close md5 file '], $self->md5_file, q[']);
+    return $md5->hexdigest;
+}
+
+
+=head2 outdir
+
+Where to write data stripped file prior to loading to iRODS
+
+=cut
+
+has 'outdir' => (
+    isa        => 'Str',
+    is         => 'ro',
+    required   => 0,
+    default    => tempdir( CLEANUP => 1 ),
+    documentation => 'Output directory for header/comment file. Default is a directory in /tmp'
+);
+
+=head2 outfile
+
+Locally written file
+
+=cut
+
+has 'outfile' => (
+         isa        => 'Str',
+         is         => 'ro',
+         required   => 0,
+         lazy_build => 1,
+);
+sub _build_outfile {
+    my $self = shift;
+    my $filename = fileparse($self->file);
+    my $outfile     = $self->outdir.qq[/$filename];
+return $outfile;
+}
+
+
+
+=head2 process
+
+Processes files for samples where consent has been withdrawn.
+
+=cut
+
+
+
+sub process {
+  my $self = shift;
+
+  $self->dry_run and $self->info('DRY RUN - no data removal');
+
+    ###check file suffix
+   if ($self->file =~ /[b|cr]am$/sxm){
+      $self->_write_header($self->_generate_header());
+    } else {
+      $self->_write_stub_file();
+    }
+     $self->_write_md5_file();
+
+     if ($self->dry_run){ carp q[Would be re-loading file ], $self->file,qq[\n] }
+     else { $self->_reload_file(); }
+
+  return $self->file;
+}
+
+sub _generate_header{
+    my $self = shift;
+
+    my $obj = WTSI::NPG::HTS::Illumina::AlnDataObject->new($self->irods,$self->file);# sub header
+
+       $obj->is_present or $self->info($self->file ,q[ not found]);
+
+    my $header = $obj->header;
+return $header;
+}
+
+
+sub _write_header{
+    my $self = shift;
+    my $header = shift;
+    push @{$header}, $DELETION_MSG;
+    if ($self->dry_run){ carp ("Would be writing header and \"$DELETION_MSG\" to ",$self->outfile) ; return }
+    write_file($self->outfile,map { "$_\n" } @{$header}) or $self->logcroak(q[Cannot write ] ,$self->outfile);
+return 1;
+}
+
+sub _write_stub_file{
+    my $self = shift;
+    if ($self->dry_run){ carp ("Would be writing \"$DELETION_MSG\" to ",$self->outfile) ; return }
+    write_file($self->outfile, $DELETION_MSG) or $self->logcroak(q[cannot write ],$self->outfile);
+return 1;
+}
+
+sub _write_md5_file{
+    my $self = shift;
+    if ($self->dry_run){ carp (q[Would be generating md5 file ],$self->md5_file);return }
+    return $self->_build_md5sum();
+}
+
+sub _reload_file {
+    my $self = shift;
+
+    ## Publisher does forced overwrites which should affect all replicates
+
+    my $publisher = WTSI::NPG::iRODS::Publisher->new(irods => $self->irods);
+
+    ##returns WTSI::NPG::iRODS::DataObject
+    my $obj =$publisher->publish_file($self->outfile,$self->file);
+
+     ## this also adds a target_history avu
+     $obj->supersede_avus('target','0');
+
+     if ($self->rt_ticket){ $obj->add_avu('rt_ticket',$self->rt_ticket) }
```

Comment:
If I comment out this code, the tests still pass. I don't think the tests cover checking that the uploaded file is correct.